src/transformers/pick.ts

```diff
@@ -0,0 +1,58 @@
+import { Constructor, Target } from '@hippo-oss/dto-decorators';
+import { plainToClass, Exclude } from 'class-transformer';
+
+/* Return the available keys for a class.
+ *
+ * Because class members are assigned as part of the `constructor` function and TypeScript
+ * metadata is lost at runtime (unless `reflect-metadata` or similar is used), the only
+ * way to enumerate a class's members is to instantiate it, especially as `class-transformer`
+ * opts to implement its own `MetadataStorage` using a private/inaccessible implementation.
+ */
+function keysForClass<T extends Target>(cls: Constructor<T>): string[] {
+    // create a canary instance of the class so we can detect its fields
+    const canary = plainToClass(cls, {});
+
+    return Reflect.ownKeys(canary).filter(
+        (key: string | symbol): key is string => typeof key === 'string',
+    );
+}
+
+/* Narrow class-transformer properties by picking the provided fields.
+ */
+export function omitTransformerProperties<T, F extends keyof T>(
```

Comment:
Just to check, is this meant to be part of the public API for this library? Noticed it isn't used here.

Reply:
Copy pasta. I think only `pick` should be supported; the `omit` API has never caught on.